### PR TITLE
Add docs about deployment strategy for data durability

### PR DESCRIPTION
**Pull request by leplatrem**

### CODE REVIEW

File: docs/deployment.rst [hunk 1]
```diff
@@ -0,0 +1,208 @@
+.. _deployment:
+
+Deployment
+##########
+
+*Kinto* is a python Web application that provides storage as a service.
+
+It relies on 3 vital components:
+* A Web stack;
```
Comment (Natim) [inline docs/deployment.rst]: A blank line is missing in order to render a rst list.

File: docs/deployment.rst [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+Deploy a PostgreSQL cluster:
+* a leader («*master*»);
+* one or more replication followers («*slaves*»).
+* A load balancer, that routes queries to take advantage of the cluster (pgPool)
+
+Writes are sent to the master, and reads are sent to the master and slaves that up-to-date.
+
+Vertical scaling:
+* Increase size of nodes (RAM+#CPU)
```
Comment (Natim) [inline docs/deployment.rst]: missing blank line

File: docs/deployment.rst [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+* Increase shared_buffers and work_mem
+
+Horizontal scaling:
+* Increase number of nodes
```
Comment (Natim) [inline docs/deployment.rst]: missing blank line

File: docs/deployment.rst [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+-----------
+
+* RAID
+* Volatile data on SSD (indexes)
+* Storage on HDD
+* shared_buffers is like caching tables in memory
+* work_mem is like caching joins (per connection)
+
+Connection pooling:
+* via load balancer
```
Comment (Natim) [inline docs/deployment.rst]: missing blank line

File: docs/deployment.rst [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+* via Kinto
+
+
+Fail safe
+---------
+
+If the master fails, one slave can be promoted to be the new master.
+
+Database crash:
+* Restore database from last scheduled backup
```
Comment (Natim) [inline docs/deployment.rst]: missing blank line

File: docs/deployment.rst [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+* automatic refresh of connections (TODO in Kinto)
+
+
+Sharding
+--------
+
+* Use buckets+collections or userid to shard ?
+
+Via pgPool:
+* Flexible
```
Comment (Natim) [inline docs/deployment.rst]: missing blank line

File: docs/deployment.rst [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+* Tedious to configure
+
+Via Kinto code:
+* not implemented yet
```
Comment (Natim) [inline docs/deployment.rst]: missing blank line

File: docs/deployment.rst [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+
+
+Fail safe
+---------
+
+WSGI process crash:
+
+* 503 error + ``Retry-After`` response header
+* Sentry report
+* Uwsgi respawns a process (or Systemd?)
```
Comment (Natim) [inline docs/deployment.rst]: Yes systemd restarts uwsgi and uwsgi handles the worker pre-fork model

File: docs/deployment.rst [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+
+See next section about details for database.
+
+
+Configuration change
+--------------------
+
+Application:
+
+# Modify configuration file
```
Comment (Natim) [inline docs/deployment.rst]: This is not detect as being a list. We should add a dot like: ``#.``

File: docs/deployment.rst [hunk 10]
```diff
@@ -0,0 +1,208 @@
...
+# Reload workers gracefully
+
+
+Database
+========
+
+*Kinto* can be configured to persist data in several kinds of storage.
+
+*PostgreSQL* is the one that we chose at Mozilla, mainly because:
+* It is a mature and standard solution;
```
Comment (Natim) [inline docs/deployment.rst]: missing blank line

File: docs/deployment.rst [hunk 11]
```diff
@@ -0,0 +1,208 @@
...
+
+Via Kinto code:
+* not implemented yet
+* battery-included (via INI configuration)
+
+
+Using Amazon RDS
+----------------
+
+*TODO*
```
Comment (Natim) [inline docs/deployment.rst]: - Basically use of Postgresql RDS (Consistency/Availability/Durability) are handled by AWS
- Use of Elasticcache for Redis
- Use of EC2 Instance with uwsgi and nginx deployed
- Use of Route53 as a loadbalancer

**Comment by leplatrem** [comment]
Thanks for your feedback ! Will update tomorrow :)

**Pull request merged**